# Starbucks reviews ☕️☕️


1. Import pandas

In [1]:
import pandas as pd

2. Read the data from starbucks_reviews.csv

In [3]:
df = pd.read_csv("starbucks_reviews.csv")

df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review                date  
0  Amber and LaDonna at the Starbucks on Southwes...  September 13, 2023  
1  ** at the Starbucks by the fire station on 436...       July 16, 2023  
2  I just wanted to go out of my way to recognize...        July 5, 2023  
3  Me and my friend were at Starbucks and my card...        May 26, 2023  
4  I’m on this kick of drinking 5 cups of warm wa...    January 22, 2023

3. How many columns/lines are in the dataset?

In [4]:
df.shape

(850, 5)

4. Convert the column "date" into datetime type.

💡 You'll have to specify the format with python's [strptime syntax](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior)

In [ ]:
df["date"] = pd.to_datetime(df["date"])

# df["date"] = pd.to_datetime(df["date"], format = "%B %d, %Y")

df.dtypes

name                object
location            object
Rating             float64
Review              object
date        datetime64[ns]
dtype: object

In [13]:
df["date"].dt.strftime("%m %d, %Y")

df["date"]

0     2023-09-13
1     2023-07-16
2     2023-07-05
3     2023-05-26
4     2023-01-22
         ...    
845   2006-07-13
846   2005-01-03
847   2004-11-05
848   2004-10-20
849   2000-07-19
Name: date, Length: 850, dtype: datetime64[ns]

5. Extract some features from the "date" column:
* year
* month
* day of week

In [ ]:
# df['year'] = df['date'].dt.year
# df['month'] = df['date'].dt.month
# df['weekday'] = df['date'].dt.dayofweek
# df.head()

df["year"], df["month"], df["weekday"] = df["date"].dt.year, df["date"].dt.month, df["date"].dt.weekday

df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  Amber and LaDonna at the Starbucks on Southwes... 2023-09-13  2023      9   
1  ** at the Starbucks by the fire station on 436... 2023-07-16  2023      7   
2  I just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  Me and my friend were at Starbucks and my card... 2023-05-26  2023      5   
4  I’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   weekday  
0        2  
1        6  
2        2  
3        4  
4        6

6. Convert the "Review" column into lowercase

In [15]:
df["Review"] = df["Review"].str.lower()

df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  amber and ladonna at the starbucks on southwes... 2023-09-13  2023      9   
1  ** at the starbucks by the fire station on 436... 2023-07-16  2023      7   
2  i just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  me and my friend were at starbucks and my card... 2023-05-26  2023      5   
4  i’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   weekday  
0        2  
1        6  
2        2  
3        4  
4        6

7. Extract features from the "Review" column:
* Count the occurrences of the word "starbucks"
* Count the total number of words in each review

In [18]:
df["count_starbucks"], df["wordcount"] = df["Review"].str.count("starbucks"), df["Review"].str.split(" ").apply(lambda x : len(x))

df.head()

name           location  Rating  \
0     Helen  Wichita Falls, TX     5.0   
1  Courtney         Apopka, FL     5.0   
2  Daynelle  Cranberry Twp, PA     5.0   
3    Taylor        Seattle, WA     5.0   
4   Tenessa        Gresham, OR     5.0   

                                              Review       date  year  month  \
0  amber and ladonna at the starbucks on southwes... 2023-09-13  2023      9   
1  ** at the starbucks by the fire station on 436... 2023-07-16  2023      7   
2  i just wanted to go out of my way to recognize... 2023-07-05  2023      7   
3  me and my friend were at starbucks and my card... 2023-05-26  2023      5   
4  i’m on this kick of drinking 5 cups of warm wa... 2023-01-22  2023      1   

   weekday  count_starbucks  wordcount  
0        2                1         60  
1        6                2        105  
2        2                1         72  
3        4                1         86  
4        6                3         74

8. Use pandas' `.describe()` method to get some basic statistics about the dataset and answer the questions below:
* Are the reviews rather good or bad?
* Are the reviews recent or old?
* Are the reviews rather long or short?
* Do people mention a lot the word "starbuck" ?

In [22]:
df.describe(include="all")

name      location      Rating          Review  \
count     850           850  705.000000             850   
unique    604           633         NaN             814   
top     Linda  New York, NY         NaN  no review text   
freq       13            14         NaN              37   
mean      NaN           NaN    1.870922             NaN   
min       NaN           NaN    1.000000             NaN   
25%       NaN           NaN    1.000000             NaN   
50%       NaN           NaN    1.000000             NaN   
75%       NaN           NaN    2.000000             NaN   
max       NaN           NaN    5.000000             NaN   
std       NaN           NaN    1.397672             NaN   

                                 date         year       month     weekday  \
count                             850   850.000000  850.000000  850.000000   
unique                            NaN          NaN         NaN         NaN   
top                               NaN          NaN         NaN         NaN   
freq                              NaN          NaN         NaN         NaN   
mean    2016-02-17 12:38:57.882352896  2015.600000    6.897647    2.955294   
min               2000-07-19 00:00:00  2000.000000    1.000000    0.000000   
25%               2012-09-26 18:00:00  2012.000000    4.000000    1.000000   
50%               2016-05-31 00:00:00  2016.000000    7.000000    3.000000   
75%               2019-01-06 06:00:00  2019.000000   10.000000    5.000000   
max               2023-09-14 00:00:00  2023.000000   12.000000    6.000000   
std                               NaN     4.258436    3.400633    1.961437   

        count_starbucks   wordcount  
count        850.000000  850.000000  
unique              NaN         NaN  
top                 NaN         NaN  
freq                NaN         NaN  
mean           1.241176   88.525882  
min            0.000000    3.000000  
25%            0.000000   50.000000  
50%            1.000000   86.000000  
75%            2.000000  123.000000  
max            9.000000  226.000000  
std            1.248584   46.477447

1. The reviews are bad, 1.87 mean, std 1.4
2. Vieilles, la moyenne est de 2015, ecart type 4.25
3. Longues, 88 de moyenne, ecart type de 46
4. Starbucks est mentionné 1.24 par avis, ce qui est bien.


The reviews are quite bad in general, as 75% of the people give a rating below 2 out of 5 !
The reviews were written between 2000 and 2023. Most of the reviews are less than 10 years old, as the median value of year is 2016.
The reviews are quite short in general, as the maximal word count is 226.
The first quartile of "count_starbucks" is 0, which means at least 25% of the people don't use the word "starbucks" at all. Then, people citing "starbucks" use it 1 or 2 times in general.
